src/main/java/com/ixortalk/organization/api/rest/OrganizationRestResource.java

```diff
@@ -48,45 +48,42 @@
 
     @Override
     @PreAuthorize("permitAll()")
-    @PostFilter("hasRole('ROLE_ADMIN') or hasRole(filterObject.role)")
+    @PostFilter("hasRole('ROLE_ADMIN') or @securityService.isAdminOfOrganization(filterObject)")
     Collection<Organization> findAll();
 
     @Override
     @PreAuthorize("permitAll()")
-    @PostFilter("hasRole('ROLE_ADMIN') or hasRole(filterObject.role)")
+    @PostFilter("hasRole('ROLE_ADMIN') or @securityService.isAdminOfOrganization(filterObject)")
     Iterable<Organization> findAll(Sort sort);
 
     @Override
     @PreAuthorize("permitAll()")
-    @Query("select o from Organization o where ?#{ hasRole('ROLE_ADMIN') } = true or o.role in ?#{ @securityService.authorities }")
+    @Query("from Organization o where ?#{ hasRole('ROLE_ADMIN') } = true or exists " +
```

Comment:
Do we need to join between organizations?

Reply:
I didn't find any other solution. The join between user organisation and its users returns a map. 
Also, we can not use the foreign key of user to org because it is not mapped in jpa.

Reply:
Something like this should work:
```    
@Query("select distinct o" +
            " from Organization o" +
            "   left outer join o.users u" +
            " where ?#{ hasRole('ROLE_ADMIN') } = true or " +
            "                (u.login = ?#{ @userEmailProvider.currentUsersEmail.orElse(null) } and u.isAdmin = true)")
```
Maybe we should also add a test case for organizations without users?  Was not relevant before, but I think it does now.

Reply:
@wjans why would it be more relevant now? What use cases would you like to be covered?

Reply:
@jraeymaek If, for instance in a client credentials flow, an organization gets created without users, that organization should still be returned when retrieving all organizations.

Reply:
Ok, will add a new empty organization in the test setup